src/components/pages/doc.js

```diff
@@ -133,13 +131,14 @@ function Doc({ initialState, PageTemplate }) {
     ? getEditDocContentRestrictionTooltip({ t, user, doc: initialState.doc, room })
     : t('editRestrictionTooltip_annonymousUser');
 
-  const [isDirty, setIsDirty] = useState(false);
+  const [view, setView] = useState(null);
```

Comment:
This makes the first render (and on the server side, the only render) have `view === null`, which fixes the control panel by not rendering it at all on the server, as we have this logic further down:

~~~
  const showHistoryPanel = view === VIEW.display || view === VIEW.history;
  const showCommentsPanel = view === VIEW.display || view === VIEW.comments;
  const showEditPanel = view === VIEW.display || view === VIEW.edit;
~~~

They are all `false` on the first render, of course.

I find it a little bit problematic though to have `view` initially basically in an undefined state. Could we maybe change it so that it picks up the initial view from the get go (not in a `useEffect`) and at the same time have a different logic (e.g. by means of `useEffect` to defer the control panel rendering until hydration has completed on the client?

Reply:
Although setting an initial state and then re-setting on re-render is a bit of double-bookkeeping with risk of inconsistency to me, I do see your point here. I will change it to have the default value, "by default".

Reply:
Why would we reset differently on re-render, we already know the correct value on the very first render: `Object.values(VIEW).find(v => v === request.query.view) || VIEW.display`? :thinking:

Reply:
Only as a left-over, because I had used `window` to get the query param and then switched back 😊 So now there's no need to re-set in `useEffect`. I have changed the logic to only set by default.

Reply:
Cool, great :+1: